src/ResourceManager/Sql/Commands.Sql/help/Add-AzureRmSqlElasticJobStep.md

```diff
@@ -0,0 +1,495 @@
+```
+
+### Add Job Step With Output Database Id Parameter Set Using Job Object
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
+ [-CommandText] <String> [-OutputDatabaseResourceId] <String> [-OutputCredentialName] <String>
+ [-OutputTableName] <String> [[-OutputSchemaName] <String>] [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobObject] <AzureSqlElasticJobModel>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Resource Id Parameter Set
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
+ [-CommandText] <String> [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobResourceId] <String>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Add Job Step With Output Database Object Parameter Set Using Job Resource Id
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
+ [-CommandText] <String> [-OutputDatabaseObject] <AzureSqlDatabaseModel> [-OutputCredentialName] <String>
+ [-OutputTableName] <String> [[-OutputSchemaName] <String>] [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobResourceId] <String>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+### Add Job Step With Output Database Id Parameter Set Using Job Resource Id
+```
+Add-AzureRmSqlElasticJobStep [-Name] <String> [-TargetGroupName] <String> [-CredentialName] <String>
+ [-CommandText] <String> [-OutputDatabaseResourceId] <String> [-OutputCredentialName] <String>
+ [-OutputTableName] <String> [[-OutputSchemaName] <String>] [-TimeoutSeconds <Int32>] [-RetryAttempts <Int32>]
+ [-InitialRetryIntervalSeconds <Int32>] [-MaximumRetryIntervalSeconds <Int32>]
+ [-RetryIntervalBackoffMultiplier <Double>] [-JobResourceId] <String>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+## DESCRIPTION
+The **Add-AzureRmSqlElasticJobStep** cmdlet adds a job step to a job
+
+## EXAMPLES
+
+### Example 1 - Adds a step to a job
+```powershell
+PS C:\> > $job | Add-AzureRmSqlElasticJobStep -Name step1 -TargetGroupName tg1 -CredentialName cred1 -CommandText "SELECT 1"
+
+JobName StepName StepId TargetGroupName CredentialName Output ExecutionOptions   CommandText
+------- -------- ------ --------------- -------------- ------ ----------------   -----------
+job1    step1    1      tg1             cred1                 (43200,10,1,120,2) SELECT 1
```

Comment:
ExecutionOptions here is not easily understandable. It should either look better by implementing .ToString() or be hidden from table output using xml file.